_grains/elasticache.py

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python
+
+import sys
+import boto.ec2
+import boto.cloudformation
+import boto.ec2.autoscale
+import boto.elasticache
+from boto.exception import BotoServerError
+import urllib2
+import time
+import random
+import logging
+
+log = logging.getLogger(__name__)
+_tags_to_grains_stack_name = 'NOT_FETCHED_YET'
+
+
+def get_aws_metadata():
+    metadata_ip = '169.254.169.254'
+    try:
+        instance_id = urllib2.urlopen(
+            'http://{0}/latest/meta-data/instance-id'.format(metadata_ip)).read().strip()
+        region = urllib2.urlopen(
+            'http://{0}/latest/meta-data/placement/availability-zone'.format(metadata_ip)).read().strip()[:-1]
+        return {'aws_instance_id': instance_id, 'aws_region': region}
+    except Exception as err:
+        sys.stderr.write('tags_to_grains ERROR: %s\n' % str(err))
+        return {'custom_grain_error': True}
```

Comment:
The dangers of copy/paste, I've also left 'courtfinder' in a function comment :)